plugins/admin/pkg/command/registry/add_test.go

```diff
@@ -102,6 +110,63 @@ func TestNewRegistryAddCommand(t *testing.T) {
 		err = json.Unmarshal(data, &r)
 		assert.NilError(t, err)
 
+		rc, ok := r.Auths["docker.io"]
+		assert.Check(t, ok)
+		assert.Equal(t, "user", rc.Username)
+		assert.Equal(t, "dummy", rc.Password)
+		assert.Equal(t, "user@default.email.com", rc.Email)
+	})
+
+	t.Run("adding registry secret success in custom namespace using custom serviceaccount", func(t *testing.T) {
+		ns := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "custom-namespace",
+			},
+		}
+		sa := corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "custom-serviceaccount",
+				Namespace: "custom-namespace",
+			},
+		}
+		client := k8sfake.NewSimpleClientset(&ns, &sa)
+		client.PrependReactor("create", "secrets", generateNameReactor)
+
+		rp := &registryAdminParams{
+			AdminParams: &pkg.AdminParams{
+				ClientSet: client,
+			},
+			Namespace:      "custom-namespace",
+			ServiceAccount: "custom-serviceaccount",
+		}
+
+		cmd := NewRegistryAddCommand(rp)
+		o, err := testutil.ExecuteCommand(cmd, "add", "--username", "user", "--password", "dummy", "--server", "docker.io")
```

Comment:
I think it's better to run the test by
```
testutil.ExecuteCommand(cmd, "add", "--username", "user", "--password", "dummy", "--server", "docker.io","--namespace","custom-namespace","--serviceaccount","custom-serviceaccount")
```
other than setting these the value of namespace and serviceaccount directly.

Reply:
Thanks, done.